### PR TITLE
View testing

### PR DESCRIPTION
**Pull request by calliminator**
Tests for MainWindow and StateManager

### CODE REVIEW

File: src_tests/manualtests/osm_configurator/view/controller_stub/settings_controller_stub.py [hunk 1]
```diff
@@ -4,8 +4,14 @@
 
 
 class SettingsControllerStub(ISettingsController):
+    def get_number_of_key_recommendations(self) -> int:
+        return 0
+
+    def set_number_of_key_recommendations(self, number_of_processes: int) -> bool:
+        return False
+
     def get_number_of_processes(self) -> int:
```
Comment (FelixWeik) [inline src_tests/manualtests/osm_configurator/view/controller_stub/settings_controller_stub.py:13]: If test behavior is fixed it's better to skip the tests i guess. Please use the pytest decorator to skip the test

File: src_tests/unittests/osm_configurator/view/states/test_state_manager.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
+import pytest
+
+import src.osm_configurator.view.states.main_window as main_window_i
+import src.osm_configurator.view.states.state_manager as state_manager_i
+import src.osm_configurator.view.states.state_name_enum as state_name_enum_i
+
+import src_tests.manualtests.osm_configurator.view.controller_stub.export_controller_stub as exportcontroller_stub_i
```
Comment (FelixWeik) [inline src_tests/unittests/osm_configurator/view/states/test_state_manager.py:7]: Should we still be using the stubs? Wouldn't it be better to use the actual implemented controllers?

File: src_tests/unittests/osm_configurator/view/states/test_state_manager.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+def main_window():
+    """Returns a legit MainWindow, which contains a legit StateManager"""
+    return main_window_i.MainWindow(export_controller=export_controller, category_controller=category_controller,
+                                    project_controller=project_controller, settings_controller=settings_controller,
+                                    aggregation_controller=aggregation_controller, calculation_controller=calculation_controller,
+                                    cut_out_controller=cut_out_controller, data_visualization_controller=data_visualization_controller,
+                                    osm_data_controller=osm_data_controller)
+
+def test_innit(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
```
Comment (FelixWeik) [inline src_tests/unittests/osm_configurator/view/states/test_state_manager.py:38]: Question: Do you want to create a new state manager for each test or do you want to use the already existing version?

File: src_tests/unittests/osm_configurator/view/states/test_state_manager.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    state_manager.change_state(state_name_enum_i.StateName.MAIN_MENU)
+
+    # The MainMenu hast no default left or right
+
+    assert not state_manager.default_go_left()
+
+    assert not state_manager.default_go_right()
+
+def test_default_go_left_and_rigth(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
```
Comment (FelixWeik) [inline src_tests/unittests/osm_configurator/view/states/test_state_manager.py:141]: Wouldn't it be better to declare the state manager once and then reuse it everytime?

**Review by FelixWeik — CHANGES_REQUESTED** [review]

File: src_tests/unittests/osm_configurator/view/states/test_state_manager.py (hunk 4 above)
Comment (GloeStoe) [reply src_tests/unittests/osm_configurator/view/states/test_state_manager.py:141]: I think it's good, that the tests are independent af each other

**Comment by calliminator** [comment]
This branch is shit, I just let it as is it, we don't merge until I got the time and mental stability to force myself to make ugly view tests work in general.

File: src_tests/unittests/osm_configurator/view/states/test_state_manager.py (hunk 3 above)
Comment (calliminator) [reply src_tests/unittests/osm_configurator/view/states/test_state_manager.py:38]: making a new one everywhere

File: src_tests/unittests/osm_configurator/view/states/test_state_manager.py (hunk 4 above)
Comment (calliminator) [reply src_tests/unittests/osm_configurator/view/states/test_state_manager.py:141]: what karl said

**Comment by calliminator** [comment]
see branch view_testing_2

**Pull request closed without merging**